app/src/main/java/org/mozilla/fenix/perf/Performance.kt

```diff
@@ -37,4 +44,22 @@ object Performance {
             activity.reportFullyDrawn()
         }
     }
+
+    fun processIntentIfPerformanceTest(intent: Intent, context: Context) {
+
+        if (intent.getBooleanExtra(HomeActivity.EXTRA_IS_PERFORMANCE_TEST, false)) {
+            val batteryStatus = context.registerReceiver(null, IntentFilter(Intent.ACTION_BATTERY_CHANGED))
+            val isPhonePlugged = batteryStatus!!.getIntExtra(BatteryManager.EXTRA_PLUGGED, -1)
```

Comment:
It's entirely possible batteryStatus returns null: can we handle the null case? Normally, I'd say wave it because this is test code but it's possible another app can trigger this maliciously so I think we should handle it.